test/e2e/cluster_samples_operator_test.go

```diff
@@ -1010,13 +1018,15 @@ func TestUpgrade(t *testing.T) {
 		t.Fatalf("problem updating deployment env")
 	}
 
-	err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
-		cfg := verifyOperatorUp(t)
-		if cfg.ConditionTrue(samplesapi.MigrationInProgress) {
-			return true, nil
-		}
-		return false, nil
-	})
+	if cfg.Status.ManagementState == operatorsv1api.Managed {
+		err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+			cfg := verifyOperatorUp(t)
+			if cfg.ConditionTrue(samplesapi.MigrationInProgress) {
```

Comment:
Potential race condition here if the migration takes less than 1 second?

Reply:
Good question.  After reviewing the code paths, I think we can confidently say the potential for a race condition is nil.  Here is my thinking ... between the update of the env above and this check, the following needs to occur, in the following order:
- the existing pod for the modified deployment needs to be terminated
- the new pod/container needs to be created / started
- version mismatch detected, MigrationInProgress is set to true
- imagestream/templates content read from disk, samples updated, image changes in progress set to true
- imagesream imports have to complete
- image changes in progress set to false 
- migration in progress set to false

I've never seen the old pod terminate / new pod created / new pod started take less than a second, much less that whole list.

Agree?